src/managers/MessageManager.js

```diff
@@ -1,39 +1,45 @@
 'use strict';
 
-const DataStore = require('./DataStore');
-const Collection = require('../util/Collection');
+const BaseManager = require('./BaseManager');
 const Message = require('../structures/Message');
+const LimitedCollection = require('../util/LimitedCollection');
+const Collection = require('../util/Collection');
 
 /**
- * Stores messages for text-based channels.
- * @extends {DataStore}
- */
-class MessageStore extends DataStore {
+	* Manages API methods for Messages and holds their cache.
+	* @extends {BaseManager}
+	*/
+class MessageManager extends BaseManager {
   constructor(channel, iterable) {
-    super(channel.client, iterable, Message);
+    super(channel.client, iterable, Message, LimitedCollection, channel.client.options.messageCacheMaxSize);
+
+    /**
+    * The channel the messages belong to.
+    * @type {TextBasedChannel}
+    */
     this.channel = channel;
   }
 
+  /**
+  * The cache of Messages.
+  * @property {LimitedCollection<Snowflake, Message>} cache
+  * @memberof MessageManager
+  * @instance
+  */
+
```

Comment:
Unsure if this will attach properly to the method if there's a space between the doc block and the code.
```suggestion
```